### PR TITLE
Migrate DownloadGitHubReleaseV0 to node10 execution

### PR DESCRIPTION
**Pull request by KonstantinTyukalov**
**Task name**: DownloadGitHubReleaseV0 

**Description**: This PR migrates DownloadGitHubReleaseV0 to node10 execution. Also some other changes provided - check changelog.

**Changelog**:
- Removed package definitions from task tests
- Changed execution to node10
- Resolved all task vulnerabilities:
  -  https://github.com/advisories/GHSA-4rq4-32rv-6wp6 
  - https://github.com/advisories/GHSA-64g7-mvw6-v9qj
  - https://github.com/advisories/GHSA-cf4h-3jhx-xvhq
  - https://github.com/advisories/GHSA-xvch-5gv4-984h
  - https://github.com/advisories/GHSA-vh95-rmgr-6w4m
  - https://github.com/advisories/GHSA-g6ww-v8xp-vmwg
- Added typescript to task definition
- Removed `vso-node-api` package since unused
- Added useful npm scripts for task build & test
- Increased timeout for one test since in pipelines it somehow reaches 2sec  - https://dev.azure.com/mseng/PipelineTools/_build/results?buildId=18020723&view=logs&j=1da87e7b-f464-5f1a-5bb7-fc605d3cc06b&t=3c4882df-5d47-5b98-21a8-f62ccef27e1e&l=62
- Removed `typings.json` file
- Changed `minimumAgentVersion` to `2.144.0`

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) N

**Attached related issue:** (Y/N) https://github.com/microsoft/azure-pipelines-tasks/issues/16250

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected - unit tests & manual run. Works fine

### CODE REVIEW

**Review by vmapetr — APPROVED** [review]

**Review by max-zaytsev — COMMENTED** [review]
We need to fix minimumAgentVersion according to our [guide](https://dev.azure.com/mseng/AzureDevOps/_wiki/wikis/AzureDevOps.wiki/18197/Upgrading-tasks-to-Node-10-(previously-Node-14))
`"minimumAgentVersion": "2.144.0",`

**Comment by v-nmursapall** [comment]
Please give me 2-3 days time, working on other priority items for now.

File: Tasks/DownloadGitHubReleaseV0/package-lock.json [hunk 1]
```diff
@@ -367,18 +393,15 @@
           }
         },
         "@types/minimatch": {
-          "version": "2.0.29",
-          "resolved": "https://registry.npmjs.org/@types/minimatch/-/minimatch-2.0.29.tgz",
+          "version": "https://registry.npmjs.org/@types/minimatch/-/minimatch-2.0.29.tgz",
```
Comment (v-nmursapall) [inline Tasks/DownloadGitHubReleaseV0/package-lock.json:396]: shouldn't version be a number here?

Comment (KonstantinTyukalov) [reply Tasks/DownloadGitHubReleaseV0/package-lock.json:396]: @v-nmursapall  I'm not sure, that's how npm generated that 😄  Regenerating does the same thing. Used node 10.24.1 and npm  6.14.12

**Review by v-nmursapall — APPROVED** [review]
The code changes look fine, approving.

**Review by max-zaytsev — APPROVED** [review]

File: Tasks/DownloadGitHubReleaseV0/package-lock.json (hunk 1 above)
Comment (KonstantinTyukalov) [reply Tasks/DownloadGitHubReleaseV0/package-lock.json]: Regenerated package lock. now version present as number

**Review by kirill-ivlev — APPROVED** [review]

**Review by mmrazik — APPROVED** [review]

**Pull request merged**